pkg/util/hostname.go

```diff
@@ -228,24 +228,19 @@ func GetHostnameData() (HostnameData, error) {
 	// and the hostname is one of the default ones
 	if getEC2Hostname, found := hostname.ProviderCatalog["ec2"]; found {
 		log.Debug("GetHostname trying EC2 metadata...")
+
+		originalHostname := hostName
 		if ecs.IsECSInstance() || ec2.IsDefaultHostname(hostName) {
-			instanceID, err := getEC2Hostname()
+			ec2Hostname, ec2Provider, errorMsg, err := getValidEc2Hostname(getEC2Hostname)
+
 			if err == nil {
-				err = validate.ValidHostname(instanceID)
-				if err == nil {
-					hostName = instanceID
-					provider = "aws"
-				} else {
-					expErr := new(expvar.String)
-					expErr.Set(err.Error())
-					hostnameErrors.Set("aws", expErr)
-					log.Debug("EC2 instance ID is not a valid hostname: ", err)
-				}
+				hostName = ec2Hostname
+				provider = ec2Provider
```

Comment:
you could simplify by setting "aws" here and removing it from the return values of `getValidEc2Hostname`?